fitbenchmarking/utils/options.py

```diff
@@ -230,14 +230,34 @@ def __init__(self, file_name=None, results_directory: str = ""):
                                                     key)
 
         fitting = config['FITTING']
-        self.num_runs = self.read_value(fitting.getint, 'num_runs')
-        self.algorithm_type = self.read_value(
-            fitting.getlist, 'algorithm_type')
-        self.software = self.read_value(fitting.getlist, 'software')
-        self.jac_method = self.read_value(fitting.getlist, 'jac_method')
+
+        if 'num_runs' in additional_options and additional_options['num_runs']:
+            self.num_runs = additional_options['num_runs']
+        else:
+            self.num_runs = self.read_value(fitting.getint, 'num_runs')
+
+        if 'algorithm_type' in additional_options and additional_options['algorithm_type']:
```

Comment:
This is good but I'm a bit concerned that typos could cause hard to detect errors. For example running the following produces no output:
```
fitbenchmarking -a levenburg-marqadt
```
whereas
```
fitbenchmarking -a levenberg-marquardt
```
works fine.

Lets have a think of how best to do this and discus it later this week, if we can use the check that's already in `Options.read_value()` it may be neat.

Reply:
Done for all except `make_plots`